src/main/utils/utils.ts

```diff
@@ -82,3 +82,23 @@ export const getObjectVariation = (original: {[key: string]: any}, updated: {[ke
 
   return variation;
 };
+
+export const convertToArray = (value: string | string[]): string[] => {
+  return Array.isArray(value) ? value : value.split(/\r?\n/);
+};
+
+export const findAddedElements = (existingElements: string[], newElements: string[]): string[] => {
+  const addedElements: string[] = [];
+  newElements
+    .filter(e => !existingElements.includes(e))
+    .forEach(e => addedElements.push(e));
+  return addedElements;
+};
+
+export const findRemovedElements = (existingElements: string[], newElements: string[]): string[] => {
+  const removedElements: string[] = [];
+  existingElements
+    .filter(e => !newElements.includes(e))
+    .forEach(e => removedElements.push(e));
+  return removedElements;
+};
```

Comment:
Aren't these the same methods, just with their variables flipped round? Couldn't you instead do a method called 'findDifferentElements' that just checked the difference from Array A to Array B, and then you flip the passed arrays when calling the function?


```
export const findDifferentElements = (arrayA: string[], arrayB: string[]): string[] => {
    const differentElements: string[] = [];
    arrayA
      .filter(e => !arrayB.includes(e))
      .forEach(e => differentElements.push(e));
    return differentElements;
  };
  
  const added = findDifferentElements(newElements, existingElements);
  const removed = findDifferentElements(existingElements, newElements);
```

Also the way this is written can just be cut down to 
```
const findRemovedElements = (existingElements: string[], newElements: string[]): string[] => {
    return existingElements.filter(e => !newElements.includes(e))
  };
```